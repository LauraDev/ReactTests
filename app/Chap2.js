import React, {Component} from 'react';
import {Image} from 'react-native';
import NY1 from './../assets/NY1.jpg'

// Afficher une image
export default class Chap2 extends Component {
    render() {
        let pic = { uri: this.props.uri};
        return <Image source={pic} style= {{width: 150, height:150}}/>
        // return <Image source={NY1} style= {{width: 150, height:150}}/>
        // return <Image source={require('./../assets/NY1.jpg')} style= {{width: 150, height:150}}/>
    }

}