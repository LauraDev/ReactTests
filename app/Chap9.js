import React, {Component} from 'react';
import {ScrollView, Text, Dimensions, FlatList} from 'react-native';
import SearchComponent from './SearchComponent';

// Fabriquer un composant de saisi pour conditionner l'appel à l'API
// Maitre en oeuvre un composant controllé
// Découpler et faire communiquer les composants

const {width, height} = Dimensions.get('window');

export default class Chap8 extends Component {

        state = {
            nbr: 0,
            term: 'play',
            retrieved: []
        }


    getMoviesFromApiAsync() {
        let url = 'https://itunes.apple.com/search?term='+this.state.term
        fetch(url)
            .then((response) => response.json())
            .then((responseJson) => {
            this.setState({nbr: responseJson.resultCount, retrieved: responseJson.results});
            })
            .catch((error) => {
            console.error(error);
        });
    }

    search = (searched) => {
        if (searched.length > 0) {
            this.setState({term: searched}, () => this.getMoviesFromApiAsync());
        }
    }
    
    render() {
        return (
            <ScrollView>
                <SearchComponent search={this.search}/>
                {this.state.nbr !== 0 && <Text>{this.state.nbr}</Text>}
                { this.state.retrieved.length > 0 && <FlatList 
                    style={{width:width}} 
                    data={this.state.retrieved}
                    keyExtractor={(item, index) => `item${index}`} 
                    renderItem={({item}) => <Text>{item.artistName}</Text>}
                />
                }
            </ScrollView>
        )
    }

}