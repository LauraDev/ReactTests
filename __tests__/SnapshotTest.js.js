import React from 'react';
import renderer from 'react-test-renderer';
import Chap2 from '../app/Chap2';
import Chap9 from '../app/Chap9';

// jest.mock('react-native-camera', () => 'camera');

test('renders correctly', () => {
    const tree = renderer.create(<Chap2 />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<Chap9 />).toJSON();
    expect(tree).toMatchSnapshot();
});