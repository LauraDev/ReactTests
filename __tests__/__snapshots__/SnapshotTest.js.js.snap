// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={
    Object {
      "uri": undefined,
    }
  }
  style={
    Object {
      "height": 150,
      "width": 150,
    }
  }
/>
`;

exports[`renders correctly 2`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 20,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Vous recherchez ...."
        style={
          Object {
            "borderColor": "#000000",
            "borderWidth": 1,
            "height": 50,
            "padding": 10,
            "paddingLeft": 5,
            "width": 730,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Rechercher
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
